src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -219,11 +224,11 @@ public FireStoreDirectoryEntry retrieveByPath(
       TableServiceClient tableServiceClient,
       UUID collectionId,
       String tableName,
-      List<String> fullPaths) {
+      Set<String> fullPaths) {
     List<TableEntity> entities =
```

Comment:
I missed the chance to comment before, but it looks like there's no need to create `entities`, you can combine both stream chained call into a single chain. This should be more memory and CPU efficient than collecting the result into an intermediate `List`

Reply:
Oh yeah, good point.